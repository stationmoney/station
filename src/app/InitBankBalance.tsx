import { PropsWithChildren } from "react"
import {
  useInitialBankBalance,
  useInitialTokenBalance,
} from "data/queries/bank"
import { BankBalanceProvider } from "data/queries/bank"
import NetworkLoading from "./NetworkLoading"

const InitBankBalance = ({ children }: PropsWithChildren<{}>) => {
  const { data: bankBalance } = useInitialBankBalance()
  const { data: tokenBalance } = useInitialTokenBalance()
  // If the balance doesn't exist, nothing is worth rendering.
  if (!bankBalance) return <NetworkLoading title="Fetching balances..." />
  return (
    <BankBalanceProvider value={[...bankBalance, ...(tokenBalance ?? [])]}>
      {children}
    </BankBalanceProvider>
  )
}

export default InitBankBalance
