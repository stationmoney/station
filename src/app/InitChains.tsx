import axios from "axios"
import { ASSETS } from "config/constants"
import { WhitelistProvider } from "data/queries/chains"
import { PropsWithChildren, useEffect, useState } from "react"

type Whitelist = Record<
  string,
  {
    token: string
    symbol: string
    name: string
    icon: string
    chains: string[]
    decimals: number
  }
>

const InitChains = ({ children }: PropsWithChildren<{}>) => {
  const [data, setData] = useState<Whitelist>()
  useEffect(() => {
    ;(async () => {
      const { data: whitelist } = await axios.get("/station/coins.json", {
        baseURL: ASSETS,
      })
      setData(whitelist)
    })()
  }, [])

  if (!data) return null

  return <WhitelistProvider value={data}>{children}</WhitelistProvider>
}

export default InitChains
