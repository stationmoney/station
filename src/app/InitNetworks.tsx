import { PropsWithChildren, useEffect, useState } from "react"
import axios from "axios"
import { ASSETS } from "config/constants"
import createContext from "utils/createContext"
import NetworkLoading from "./NetworkLoading"

export const [useNetworks, NetworksProvider] = createContext<{
  networks: InterchainNetworks
  filterEnabledNetworks: <T>(network: Record<string, T>) => Record<string, T>
}>("useNetworks")

const InitNetworks = ({ children }: PropsWithChildren<{}>) => {
  const [networks, setNetworks] = useState<InterchainNetworks>()
  const [enabledNetworks, setEnabledNetworks] = useState<string[]>([])

  useEffect(() => {
    const fetchChains = async () => {
      const { data: chains } = await axios.get<InterchainNetworks>(
        "/station/chains.json",
        {
          baseURL: ASSETS,
        }
      )
      setNetworks(chains)
    }

    fetchChains()
  }, [])

  useEffect(() => {
    const testChains = async () => {
      if (!networks) return
      const testBase = {
        ...networks.mainnet,
        ...networks.testnet,
        ...networks.classic,
      }

      const result = await Promise.all(
        Object.values(testBase).map(async (network) => {
          try {
            const { data } = await axios.get(
              "cosmos/base/tendermint/v1beta1/node_info",
              {
                baseURL: network.lcd,
                timeout: 5_000,
              }
            )
            return data.default_node_info.network as string
          } catch (e) {
            return null
          }
        })
      )

      setEnabledNetworks(
        result.filter((r) => typeof r === "string") as string[]
      )
    }

    testChains()
  }, [networks])

  if (!networks || !enabledNetworks.length)
    return <NetworkLoading title="Connecting to available networks..." />

  return (
    <NetworksProvider
      value={{
        networks,
        filterEnabledNetworks: (networks) =>
          Object.fromEntries(
            Object.entries(networks).filter(([chainID]) =>
              enabledNetworks.includes(chainID)
            )
          ),
      }}
    >
      {children}
    </NetworksProvider>
  )
}

export default InitNetworks
