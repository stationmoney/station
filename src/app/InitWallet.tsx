import { PropsWithChildren, useEffect, useMemo } from "react"
import { QueryClient, QueryClientProvider } from "react-query"
import { WalletStatus } from "@terra-money/wallet-types"
import { useWallet } from "@terra-money/use-wallet"
import { useNetworkName } from "data/wallet"
import { isWallet, useAuth } from "auth"
import Online from "./containers/Online"
import NetworkLoading from "./NetworkLoading"

const InitWallet = ({ children }: PropsWithChildren<{}>) => {
  useOnNetworkChange()
  const { status } = useWallet()
  const queryClient = useQueryClient()
  const networkName = useNetworkName()

  return status === WalletStatus.INITIALIZING ? (
    <NetworkLoading title="Initializing your wallet..." />
  ) : (
    <QueryClientProvider client={queryClient} key={networkName}>
      {children}
      <Online />
    </QueryClientProvider>
  )
}

export default InitWallet

/* hooks */
const useOnNetworkChange = () => {
  const { wallet, disconnect } = useAuth()
  const isPreconfiguredWallet = isWallet.preconfigured(wallet)
  const shouldDisconnect = isPreconfiguredWallet

  useEffect(() => {
    if (shouldDisconnect) disconnect()
  }, [disconnect, shouldDisconnect])
}

const useQueryClient = () => {
  const name = useNetworkName()

  return useMemo(() => {
    if (!name) throw new Error()
    return new QueryClient()
  }, [name])
}
