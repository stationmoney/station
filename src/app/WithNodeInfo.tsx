import { PropsWithChildren } from "react"
import { useNodeInfo } from "data/queries/tendermint"
import Overlay from "./components/Overlay"
import NetworkError from "./NetworkError"
import { useNetworkName } from "data/wallet"

const WithNodeInfo = ({ children }: PropsWithChildren<{}>) => {
  const network = useNetworkName()
  // FIXME: remove or make sure chainID is available
  const { isLoading, isError } = useNodeInfo(
    network === "mainnet" ? "phoenix-1" : "pisco-1"
  )

  if (isError) {
    return (
      <Overlay>
        <NetworkError />
      </Overlay>
    )
  }

  if (isLoading) return null
  return <>{children}</>
}

export default WithNodeInfo
