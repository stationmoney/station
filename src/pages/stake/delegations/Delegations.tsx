import { useTranslation } from "react-i18next"
import { AccAddress, Coin, Delegation } from "@terra-money/feather.js"
import { getMaxHeightStyle } from "utils/style"
import { combineState } from "data/query"
import { useMemoizedPrices } from "data/queries/coingecko"
import { useInterchainDelegations } from "data/queries/staking"
import { ValidatorLink } from "components/general"
import { ModalButton } from "components/feedback"
import { Table } from "components/layout"
import { Read } from "components/token"
import StakedCard from "../components/StakedCard"
import { useNativeDenoms } from "data/token"
import styles from "../CardModal.module.scss"

const Delegations = () => {
  const { t } = useTranslation()
  const readNativeDenom = useNativeDenoms()
  const { data: prices, ...pricesState } = useMemoizedPrices()

  const interchainDelegations = useInterchainDelegations()

  const delegations: Delegation[] = interchainDelegations.reduce(
    (acc, { data }) => (data ? [...data?.delegation, ...acc] : acc),
    [] as Delegation[]
  )
  const state = combineState(pricesState, ...interchainDelegations)

  /* render */
  const title = t("Delegations")

  const render = () => {
    if (!delegations || !prices) return null

    const total = delegations.reduce((acc, { balance }) => {
      const { token, decimals } = readNativeDenom(balance.denom)
      return (
        acc +
        (balance.amount.toNumber() * (prices[token]?.price || 0)) /
          10 ** decimals
      )
    }, 0)

    return (
      <ModalButton
        title={title}
        renderButton={(open) => (
          <StakedCard
            {...state}
            title={
              <div className={styles.header_wrapper}>
                {title}
                {total !== -1 && (
                  <span className={styles.view_more}>View More</span>
                )}
              </div>
            }
            value={total?.toString() || "0"}
            onClick={open}
            cardName={"delegations"}
          />
        )}
      >
        <Table
          dataSource={delegations}
          sorter={({ balance: { amount: a } }, { balance: { amount: b } }) =>
            b.minus(a).toNumber()
          }
          columns={[
            {
              title: t("Validator"),
              dataIndex: "validator_address",
              render: (address: AccAddress) => (
                <ValidatorLink address={address} internal />
              ),
            },
            {
              title: t("Delegated"),
              dataIndex: "balance",
              render: (balance: Coin) => <Read {...balance.toData()} />,
              align: "right",
            },
          ]}
          style={getMaxHeightStyle(320)}
        />
      </ModalButton>
    )
  }

  return render()
}

export default Delegations
