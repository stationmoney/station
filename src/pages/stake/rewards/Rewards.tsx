import { useTranslation } from "react-i18next"
import { useCurrency } from "data/settings/Currency"
import { combineState } from "data/query"
import { calcRewardsValues, useRewards } from "data/queries/distribution"
import { useExchangeRates, useMemoizedPrices } from "data/queries/coingecko"
import { useMemoizedCalcValue } from "data/queries/coingecko"
import { useNativeDenoms, WithTokenItem } from "data/token"
import { ModalButton } from "components/feedback"
import { TokenCard, TokenCardGrid } from "components/token"
import { TooltipIcon } from "components/display"
import StakedCard from "../components/StakedCard"
import RewardsTooltip from "../RewardsTooltip"
import styles from "../CardModal.module.scss"
import { Coin } from "@terra-money/feather.js"

const Rewards = () => {
  const { t } = useTranslation()
  const currency = useCurrency()
  const calcValue = useMemoizedCalcValue()
  const readNativeDenom = useNativeDenoms()

  const { data: prices, ...pricesState } = useMemoizedPrices()
  const { data: rewards, ...rewardsState } = useRewards()
  const { data: exchangeRates, ...exchangeRatesState } = useExchangeRates()

  const state = combineState(rewardsState, exchangeRatesState, pricesState)

  /* render */
  const title = t("Staking rewards")
  const render = () => {
    let sameDenom: boolean = true
    const coinsValue =
      !rewards || !prices
        ? 0
        : rewards.total.toData()?.reduce((acc, { amount, denom }) => {
            const { token, decimals } = readNativeDenom(denom)
            if (denom !== rewards?.total.toData()[0].denom) {
              sameDenom = false
            }
            return (
              acc +
              (parseInt(amount) * (prices?.[token]?.price || 0)) /
                10 ** decimals
            )
          }, 0)

    const totalToDisplay = coinsValue

    let list: Coin.Data[] = []
    if (rewards) {
      const { total } = calcRewardsValues(rewards, currency.id, calcValue)
      list = total.list
    }

    const showTokens = totalToDisplay === -1 || sameDenom

    console.log({ totalToDisplay, sameDenom, showTokens })

    return (
      <ModalButton
        title={title}
        renderButton={(open) => (
          <StakedCard
            {...state}
            title={
              <div className={styles.header_wrapper}>
                <TooltipIcon content={<RewardsTooltip />} placement="bottom">
                  {title}
                </TooltipIcon>
                {!showTokens && (
                  <span className={styles.view_more}>View More</span>
                )}
              </div>
            }
            value={totalToDisplay?.toString() || "-1"}
            showTokens={showTokens}
            onClick={open}
            cardName="rewards"
          ></StakedCard>
        )}
      >
        <TokenCardGrid maxHeight>
          {list?.map(({ amount, denom }) => (
            <WithTokenItem token={denom} key={denom}>
              {(item) => (
                <TokenCard
                  {...item}
                  amount={amount}
                  value={calcValue({ amount, denom })}
                />
              )}
            </WithTokenItem>
          ))}
        </TokenCardGrid>
      </ModalButton>
    )
  }

  return render()
}

export default Rewards
