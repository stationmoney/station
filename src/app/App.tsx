import { getErrorMessage } from "utils/error"
import Layout, { MainContainer, Page } from "components/layout"
import { Banner, Content, Header, Actions, Sidebar } from "components/layout"
import { ErrorBoundary, Wrong } from "components/feedback"

/* routes */
import { useNav } from "./routes"

/* banner */
import UpdateExtension from "./sections/UpdateExtension"

/* sidebar */
import Nav from "./sections/Nav"
import Aside from "./sections/Aside"

/* header */
import NetworkHeader from "./sections/NetworkHeader"
import Refresh from "./sections/Refresh"
import Preferences from "./sections/Preferences"
import ConnectWallet from "./sections/ConnectWallet"

/* extra */
import LatestTx from "./sections/LatestTx"
import ValidatorButton from "./sections/ValidatorButton"
import DevTools from "./sections/DevTools"
import { useShowWelcomeModal } from "utils/localStorage"

/* init */
import InitBankBalance from "./InitBankBalance"
import Wallet from "pages/wallet/Wallet"
import WelcomeModal from "./sections/WelcomeModal"
import NavButton from "./sections/NavButton"
import NetworkStatus from "components/display/NetworkStatus"

const App = () => {
  const { element: routes } = useNav()
  const showModal = useShowWelcomeModal()

  return (
    <Layout>
      <Banner>
        <UpdateExtension />
      </Banner>

      <Sidebar>
        <Nav />
        <Aside />
      </Sidebar>

      <Header>
        <NetworkHeader />

        <Actions>
          <DevTools />
          <section>
            <Refresh />
            <Preferences />
            <NetworkStatus />
          </section>
          <ValidatorButton />
          <ConnectWallet />
          <NavButton />
        </Actions>
        <LatestTx />
      </Header>

      <Content>
        <ErrorBoundary fallback={fallback}>
          <InitBankBalance>
            <MainContainer>
              {routes}
              <Wallet />
              {showModal && <WelcomeModal />}
            </MainContainer>
          </InitBankBalance>
        </ErrorBoundary>
      </Content>
    </Layout>
  )
}

export default App

/* error */
export const fallback = (error: Error) => (
  <Page>
    <Wrong>{getErrorMessage(error)}</Wrong>
  </Page>
)
