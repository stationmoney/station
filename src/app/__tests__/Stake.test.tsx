import React from "react"
import { render, screen } from "@testing-library/react"
import Stake from "pages/stake/Stake"
import { NetworksProvider } from "app/InitNetworks"
import { RecoilRoot } from "recoil"
import { MemoryRouter } from "react-router-dom"
import { mockNetworks } from "./__mocks__/Networks.mock"
import { mockValidators } from "./__mocks__/Validators.mock"
import { mockDelegations } from "./__mocks__/Delegations.mock"
import { mockCalcDelegations } from "./__mocks__/CalcDelegations.mock"
import { mockUnbondings } from "./__mocks__/Unbondings.mock"
import { mockRewards } from "./__mocks__/Rewards.mock"
import { mockBalances } from "./__mocks__/Balances.mock"
import user from "@testing-library/user-event"

function renderComponent() {
  type TokenFilter = <T>(network: Record<string, T>) => Record<string, T>
  const networks = {} as jest.Mocked<InterchainNetworks>
  const mockedTokenFilter = () => true as jest.Mocked<TokenFilter>

  return render(
    <NetworksProvider
      value={{
        networks: networks,
        filterEnabledNetworks: mockedTokenFilter,
        filterDisabledNetworks: mockedTokenFilter,
      }}
    >
      <MemoryRouter>
        <RecoilRoot>
          <Stake />
        </RecoilRoot>
      </MemoryRouter>
    </NetworksProvider>
  )
}

jest.mock("../../data/wallet", () => {
  const mockUseNetwork = () => {
    return mockNetworks
  }

  const mockUseChainID = () => {
    return "pisco-1"
  }

  return {
    useNetwork: mockUseNetwork,
    useChainID: mockUseChainID,
  }
})

jest.mock("@terra-money/wallet-provider", () => {
  const actual = jest.requireActual("@terra-money/wallet-provider")

  const useWallet = () => {
    return {
      status: "WALLET_CONNECTED",
      network: ["pisco-1"],
    }
  }

  return {
    ...actual,
    useWallet: useWallet,
  }
})

jest.mock("../../data/queries/staking", () => {
  const mockUseInterchainDelegations = () => mockDelegations
  const mockUseInterchainValidators = () => mockValidators
  const mockUseCalcDelegationsByValidator = () => mockCalcDelegations
  const mockUseDelegations = () => mockDelegations
  const mockUseUnbondings = () => mockUnbondings
  const mockUseValidators = () => mockValidators

  return {
    useInterchainDelegations: mockUseInterchainDelegations,
    useInterchainValidators: mockUseInterchainValidators,
    useCalcDelegationsByValidator: mockUseCalcDelegationsByValidator,
    useDelegations: mockUseDelegations,
    useUnbondings: mockUseUnbondings,
    useValidators: mockUseValidators,
  }
})

jest.mock("../../data/queries/distribution", () => {
  const mockUseRewards = () => {
    return mockRewards
  }

  return {
    useRewards: mockUseRewards,
  }
})

jest.mock("../../data/queries/bank", () => {
  const mockUseBalances = () => {
    return mockBalances
  }

  return {
    useBalances: mockUseBalances,
  }
})

describe("Stake page component matches snapshots", async () => {
  it("matches original component", () => {
    const { asFragment } = renderComponent()
    expect(asFragment()).toMatchSnapshot()
  })

  it("matches undelegate page", async () => {
    const { asFragment } = renderComponent()

    const undelegateButton = screen.getByRole("button", {
      name: /Undelegate/i,
    })
    await user.click(undelegateButton)

    expect(asFragment()).toMatchSnapshot()
  })

  it("matches manual staking page", async () => {
    const { asFragment } = renderComponent()

    const manualStakingButton = screen.getByRole("button", {
      name: /Manual Stake/i,
    })
    await user.click(manualStakingButton)

    expect(asFragment()).toMatchSnapshot()
  })

  it("matches manual staking page with search", async () => {
    const { asFragment } = renderComponent()

    const manualStakingButton = screen.getByRole("button", {
      name: /Manual Stake/i,
    })
    await user.click(manualStakingButton)

    const searchBar = screen.getByRole("textbox")
    await user.click(searchBar)
    await user.keyboard("Terran")

    expect(asFragment()).toMatchSnapshot()
  })
})
