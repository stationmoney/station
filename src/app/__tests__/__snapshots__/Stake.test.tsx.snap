// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Stake page component matches snapshots matches manual staking page 1`] = `
<DocumentFragment>
  <article
    class="page"
  >
    <div
      class="container grid"
    >
      <header
        class="header"
      >
        <div
          class="titleWrapper"
        >
          <h1
            class="title"
          >
            Stake
          </h1>
        </div>
        <a
          class="button small primary"
          href="/rewards"
        >
          Withdraw all rewards
        </a>
      </header>
      <section
        class="main"
      >
        <div
          class="col"
          style="flex: 1;"
        >
          <div
            class="tabsContainer"
          >
            <section
              class="tabs page"
            >
              <button
                class="tab"
                type="button"
              >
                <div
                  class="title"
                >
                  Quick Stake
                </div>
                <span
                  class="inline start"
                  style="gap: 4px;"
                >
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit muted css-1vooibu-MuiSvgIcon-root"
                    data-testid="HelpOutlineOutlinedIcon"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M11 18h2v-2h-2v2zm1-16C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.41 0-8-3.59-8-8s3.59-8 8-8 8 3.59 8 8-3.59 8-8 8zm0-14c-2.21 0-4 1.79-4 4h2c0-1.1.9-2 2-2s2 .9 2 2c0 2-3 1.75-3 5h2c0-2.25 3-2.5 3-5 0-2.21-1.79-4-4-4z"
                    />
                  </svg>
                </span>
              </button>
              <button
                class="tab active"
                type="button"
              >
                <div
                  class="title"
                >
                  Manual Stake
                </div>
              </button>
            </section>
            <div
              class="content"
            >
              <article
                class="page sub"
              >
                <div
                  class="container grid"
                >
                  <section
                    class="main"
                  >
                    <div
                      class="grid"
                      style="gap: 0;"
                    >
                      <div
                        class="wrapper search"
                      >
                        <input
                          autocomplete="off"
                          class="input"
                          inputmode="search"
                          placeholder="Search for validator..."
                          value=""
                        />
                        <svg
                          aria-hidden="true"
                          class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium icon css-i4bv87-MuiSvgIcon-root"
                          data-testid="SearchIcon"
                          focusable="false"
                          viewBox="0 0 24 24"
                        >
                          <path
                            d="M15.5 14h-.79l-.28-.27C15.41 12.59 16 11.11 16 9.5 16 5.91 13.09 3 9.5 3S3 5.91 3 9.5 5.91 16 9.5 16c1.61 0 3.09-.59 4.23-1.57l.27.28v.79l5 4.99L20.49 19l-4.99-5zm-6 0C7.01 14 5 11.99 5 9.5S7.01 5 9.5 5 14 7.01 14 9.5 11.99 14 9.5 14z"
                          />
                        </svg>
                      </div>
                      <div
                        class="chainfilter__out"
                      >
                        <div
                          class="filter header"
                        >
                          <div
                            class="pills"
                          >
                            <button
                              class="active"
                            >
                              <img
                                alt="Terra"
                                src="https://station-assets.terra.money/img/chains/Terra.svg"
                              />
                              Terra
                            </button>
                            <button>
                              <img
                                alt="Atreides"
                                src="https://station-assets.terra.money/img/chains/Atreides.png"
                              />
                              Atreides
                            </button>
                            <button>
                              <img
                                alt="Corrino"
                                src="https://station-assets.terra.money/img/chains/Corrino.png"
                              />
                              Corrino
                            </button>
                            <button>
                              <img
                                alt="Harkonnen"
                                src="https://station-assets.terra.money/img/chains/Harkonnen.png"
                              />
                              Harkonnen
                            </button>
                            <button>
                              <img
                                alt="Ordos"
                                src="https://station-assets.terra.money/img/chains/Ordos.png"
                              />
                              Ordos
                            </button>
                          </div>
                        </div>
                        <div
                          class="content"
                        />
                      </div>
                    </div>
                  </section>
                </div>
              </article>
            </div>
          </div>
        </div>
      </section>
    </div>
  </article>
</DocumentFragment>
`;

exports[`Stake page component matches snapshots matches manual staking page with search 1`] = `
<DocumentFragment>
  <article
    class="page"
  >
    <div
      class="container grid"
    >
      <header
        class="header"
      >
        <div
          class="titleWrapper"
        >
          <h1
            class="title"
          >
            Stake
          </h1>
        </div>
        <a
          class="button small primary"
          href="/rewards"
        >
          Withdraw all rewards
        </a>
      </header>
      <section
        class="main"
      >
        <div
          class="col"
          style="flex: 1;"
        >
          <div
            class="tabsContainer"
          >
            <section
              class="tabs page"
            >
              <button
                class="tab"
                type="button"
              >
                <div
                  class="title"
                >
                  Quick Stake
                </div>
                <span
                  class="inline start"
                  style="gap: 4px;"
                >
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit muted css-1vooibu-MuiSvgIcon-root"
                    data-testid="HelpOutlineOutlinedIcon"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M11 18h2v-2h-2v2zm1-16C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.41 0-8-3.59-8-8s3.59-8 8-8 8 3.59 8 8-3.59 8-8 8zm0-14c-2.21 0-4 1.79-4 4h2c0-1.1.9-2 2-2s2 .9 2 2c0 2-3 1.75-3 5h2c0-2.25 3-2.5 3-5 0-2.21-1.79-4-4-4z"
                    />
                  </svg>
                </span>
              </button>
              <button
                class="tab active"
                type="button"
              >
                <div
                  class="title"
                >
                  Manual Stake
                </div>
              </button>
            </section>
            <div
              class="content"
            >
              <article
                class="page sub"
              >
                <div
                  class="container grid"
                >
                  <section
                    class="main"
                  >
                    <div
                      class="grid"
                      style="gap: 0;"
                    >
                      <div
                        class="wrapper search"
                      >
                        <input
                          autocomplete="off"
                          class="input"
                          inputmode="search"
                          placeholder="Search for validator..."
                          value="Terran"
                        />
                        <svg
                          aria-hidden="true"
                          class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium icon css-i4bv87-MuiSvgIcon-root"
                          data-testid="SearchIcon"
                          focusable="false"
                          viewBox="0 0 24 24"
                        >
                          <path
                            d="M15.5 14h-.79l-.28-.27C15.41 12.59 16 11.11 16 9.5 16 5.91 13.09 3 9.5 3S3 5.91 3 9.5 5.91 16 9.5 16c1.61 0 3.09-.59 4.23-1.57l.27.28v.79l5 4.99L20.49 19l-4.99-5zm-6 0C7.01 14 5 11.99 5 9.5S7.01 5 9.5 5 14 7.01 14 9.5 11.99 14 9.5 14z"
                          />
                        </svg>
                      </div>
                      <div
                        class="chainfilter__out"
                      >
                        <div
                          class="filter header"
                        >
                          <div
                            class="pills"
                          >
                            <button
                              class="active"
                            >
                              <img
                                alt="Terra"
                                src="https://station-assets.terra.money/img/chains/Terra.svg"
                              />
                              Terra
                            </button>
                            <button>
                              <img
                                alt="Atreides"
                                src="https://station-assets.terra.money/img/chains/Atreides.png"
                              />
                              Atreides
                            </button>
                            <button>
                              <img
                                alt="Corrino"
                                src="https://station-assets.terra.money/img/chains/Corrino.png"
                              />
                              Corrino
                            </button>
                            <button>
                              <img
                                alt="Harkonnen"
                                src="https://station-assets.terra.money/img/chains/Harkonnen.png"
                              />
                              Harkonnen
                            </button>
                            <button>
                              <img
                                alt="Ordos"
                                src="https://station-assets.terra.money/img/chains/Ordos.png"
                              />
                              Ordos
                            </button>
                          </div>
                        </div>
                        <div
                          class="content"
                        />
                      </div>
                    </div>
                  </section>
                </div>
              </article>
            </div>
          </div>
        </div>
      </section>
    </div>
  </article>
</DocumentFragment>
`;

exports[`Stake page component matches snapshots matches original component 1`] = `
<DocumentFragment>
  <article
    class="page"
  >
    <div
      class="container grid"
    >
      <header
        class="header"
      >
        <div
          class="titleWrapper"
        >
          <h1
            class="title"
          >
            Stake
          </h1>
        </div>
        <a
          class="button small primary"
          href="/rewards"
        >
          Withdraw all rewards
        </a>
      </header>
      <section
        class="main"
      >
        <div
          class="col"
          style="flex: 1;"
        >
          <div
            class="tabsContainer"
          >
            <section
              class="tabs page"
            >
              <button
                class="tab active"
                type="button"
              >
                <div
                  class="title"
                >
                  Quick Stake
                </div>
                <span
                  class="inline start"
                  style="gap: 4px;"
                >
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit muted css-1vooibu-MuiSvgIcon-root"
                    data-testid="HelpOutlineOutlinedIcon"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M11 18h2v-2h-2v2zm1-16C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.41 0-8-3.59-8-8s3.59-8 8-8 8 3.59 8 8-3.59 8-8 8zm0-14c-2.21 0-4 1.79-4 4h2c0-1.1.9-2 2-2s2 .9 2 2c0 2-3 1.75-3 5h2c0-2.25 3-2.5 3-5 0-2.21-1.79-4-4-4z"
                    />
                  </svg>
                </span>
              </button>
              <button
                class="tab"
                type="button"
              >
                <div
                  class="title"
                >
                  Manual Stake
                </div>
              </button>
            </section>
            <div
              class="content"
            >
              <article
                class="page"
              >
                <div
                  class="container grid"
                >
                  <section
                    class="main"
                  >
                    <div
                      class="buttonfilter"
                    >
                      <div
                        class="header"
                      />
                      <div
                        class="actions"
                      >
                        <button
                          class="active"
                        >
                          Delegate
                        </button>
                        <button>
                          Undelegate
                        </button>
                      </div>
                      <div
                        class="content"
                      >
                        <article
                          class="card muted default"
                        >
                          <section
                            class="main"
                          >
                            <article
                              class="page small invisible"
                            >
                              <div
                                class="container grid"
                              >
                                <section
                                  class="main"
                                >
                                  <div
                                    class="chainfilter__out"
                                  >
                                    <div
                                      class="filter header"
                                    >
                                      <div
                                        class="pills"
                                      >
                                        <button
                                          class="active"
                                        >
                                          <img
                                            alt="Terra"
                                            src="https://station-assets.terra.money/img/chains/Terra.svg"
                                          />
                                          Terra
                                        </button>
                                        <button>
                                          <img
                                            alt="Atreides"
                                            src="https://station-assets.terra.money/img/chains/Atreides.png"
                                          />
                                          Atreides
                                        </button>
                                        <button>
                                          <img
                                            alt="Corrino"
                                            src="https://station-assets.terra.money/img/chains/Corrino.png"
                                          />
                                          Corrino
                                        </button>
                                        <button>
                                          <img
                                            alt="Harkonnen"
                                            src="https://station-assets.terra.money/img/chains/Harkonnen.png"
                                          />
                                          Harkonnen
                                        </button>
                                        <button>
                                          <img
                                            alt="Ordos"
                                            src="https://station-assets.terra.money/img/chains/Ordos.png"
                                          />
                                          Ordos
                                        </button>
                                      </div>
                                    </div>
                                    <div
                                      class="content"
                                    />
                                  </div>
                                </section>
                              </div>
                            </article>
                          </section>
                        </article>
                      </div>
                    </div>
                  </section>
                </div>
              </article>
            </div>
          </div>
        </div>
      </section>
    </div>
  </article>
</DocumentFragment>
`;

exports[`Stake page component matches snapshots matches undelegate page 1`] = `
<DocumentFragment>
  <article
    class="page"
  >
    <div
      class="container grid"
    >
      <header
        class="header"
      >
        <div
          class="titleWrapper"
        >
          <h1
            class="title"
          >
            Stake
          </h1>
        </div>
        <a
          class="button small primary"
          href="/rewards"
        >
          Withdraw all rewards
        </a>
      </header>
      <section
        class="main"
      >
        <div
          class="col"
          style="flex: 1;"
        >
          <div
            class="tabsContainer"
          >
            <section
              class="tabs page"
            >
              <button
                class="tab active"
                type="button"
              >
                <div
                  class="title"
                >
                  Quick Stake
                </div>
                <span
                  class="inline start"
                  style="gap: 4px;"
                >
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit muted css-1vooibu-MuiSvgIcon-root"
                    data-testid="HelpOutlineOutlinedIcon"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M11 18h2v-2h-2v2zm1-16C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.41 0-8-3.59-8-8s3.59-8 8-8 8 3.59 8 8-3.59 8-8 8zm0-14c-2.21 0-4 1.79-4 4h2c0-1.1.9-2 2-2s2 .9 2 2c0 2-3 1.75-3 5h2c0-2.25 3-2.5 3-5 0-2.21-1.79-4-4-4z"
                    />
                  </svg>
                </span>
              </button>
              <button
                class="tab"
                type="button"
              >
                <div
                  class="title"
                >
                  Manual Stake
                </div>
              </button>
            </section>
            <div
              class="content"
            >
              <article
                class="page"
              >
                <div
                  class="container grid"
                >
                  <section
                    class="main"
                  >
                    <div
                      class="buttonfilter"
                    >
                      <div
                        class="header"
                      />
                      <div
                        class="actions"
                      >
                        <button>
                          Delegate
                        </button>
                        <button
                          class="active"
                        >
                          Undelegate
                        </button>
                      </div>
                      <div
                        class="content"
                      >
                        <article
                          class="card muted default"
                        >
                          <section
                            class="main"
                          >
                            <article
                              class="page small invisible"
                            >
                              <div
                                class="container grid"
                              >
                                <section
                                  class="main"
                                >
                                  <div
                                    class="chainfilter__out"
                                  >
                                    <div
                                      class="filter header"
                                    >
                                      <div
                                        class="pills"
                                      >
                                        <button
                                          class="active"
                                        >
                                          <img
                                            alt="Terra"
                                            src="https://station-assets.terra.money/img/chains/Terra.svg"
                                          />
                                          Terra
                                        </button>
                                        <button>
                                          <img
                                            alt="Atreides"
                                            src="https://station-assets.terra.money/img/chains/Atreides.png"
                                          />
                                          Atreides
                                        </button>
                                        <button>
                                          <img
                                            alt="Corrino"
                                            src="https://station-assets.terra.money/img/chains/Corrino.png"
                                          />
                                          Corrino
                                        </button>
                                        <button>
                                          <img
                                            alt="Harkonnen"
                                            src="https://station-assets.terra.money/img/chains/Harkonnen.png"
                                          />
                                          Harkonnen
                                        </button>
                                        <button>
                                          <img
                                            alt="Ordos"
                                            src="https://station-assets.terra.money/img/chains/Ordos.png"
                                          />
                                          Ordos
                                        </button>
                                      </div>
                                    </div>
                                    <div
                                      class="content"
                                    />
                                  </div>
                                </section>
                              </div>
                            </article>
                          </section>
                        </article>
                      </div>
                    </div>
                  </section>
                </div>
              </article>
            </div>
          </div>
        </div>
      </section>
    </div>
  </article>
</DocumentFragment>
`;
